### DISCUSSION TITLE
Quarto not detecting R inside RStudio

### DISCUSSION THREAD

**Discussion by dhodge180** [quarto-dev/quarto-cli#3707] [Q&A]
I'm in Ubuntu (tries 20.04 and 22.04). Inside RStudio quarto is not finding the R installation.
quarto check run on the command line doesn't find it *unless* I run "sudo quarto check" any ideas as to what I need to do to fix this?

I have even added the QUARTO_R environment variable to point at the directory with R and Rscript in them. R runs fine in the commandline (without sudo).

This is the output of quarto check: (run in the terminal inside rstudio)

[✓] Checking Quarto installation......OK
      Version: 1.1.189
      Path: /opt/rstudio/rstudio-2022.07.2+576/bin/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.6
      Path: /usr/bin/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

Rscript execution error: No such file or directory
[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/

The actual location of R is the same as python3 above, i.e. which tells me exectuables are at /usr/bin/R and /usr/bin/Rscript

(PS - I realize I had to install quarto externally to RStudio to run it on the command line and discover this, but I removed it after)

**Comment by D3SL** [comment]
I'm having this same problem on Windows 10. R and the quarto preview command work fine in CMD but not within rstudio itself. This extends to other libraries calling quarto like `{targets}`.

Update:

After _extensive_ hair pulling the problem was due to python's appalling dependency handling. The terminal inside RStudio inherited from the mamba python environment while a raw cmd window did not. Having `r-base` and `rpy2` installed in that mamba environment was causing quarto inside RStudio to first use the python `r-base` install of R rather than the actual system R which RStudio itself was configured to use.

I also had to reinstall jupyter for whatever reason despite the error thrown at that check being about not finding the `yaml` module.
